From 59693c0334efecc5a4eb847dd3e506946709811b Mon Sep 17 00:00:00 2001
From: ejaaskel <esa.jaaskela@suomi24.fi>
Date: Mon, 10 Nov 2025 12:11:30 +0000
Subject: [PATCH] policy/modules/system/locallogin: Allow changing password

Loosen up the SELinux rules so that the user password can be changed
during the log-in process. This requires that the local_login_t has
additional permissions on etc files and directories, and shadow files.
In addtion, the process needs to be able to perform setfscreate.
 
---
 policy/modules/system/locallogin.te | 1 +
 1 file changed, 1 insertion(+)
 
diff --git a/policy/modules/system/locallogin.te b/policy/modules/system/locallogin.te
index 4ba131d29..503e9fb6f 100644
--- a/policy/modules/system/locallogin.te
+++ b/policy/modules/system/locallogin.te
@@ -47,6 +47,9 @@ allow local_login_t self:sem create_sem_perms;
 allow local_login_t self:msgq create_msgq_perms;
 allow local_login_t self:msg { receive send };
 allow local_login_t self:key { link search write };
+allow local_login_t self:process setfscreate;
+files_manage_etc_files(local_login_t)
+auth_manage_shadow(local_login_t)
 
 allow local_login_t local_login_lock_t:file manage_file_perms;
 files_lock_filetrans(local_login_t, local_login_lock_t, file)
--
2.43.0

